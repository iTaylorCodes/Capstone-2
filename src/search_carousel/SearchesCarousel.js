import { Link } from "react-router-dom";

/** SearchesCarousel.
 *
 * Carousel component to show recent searches from Redux reducer.
 *    SearchesCarousel images link to /neighborhood/:neighborhood route.
 *
 * Rendered on homepage.
 */
const SearchesCarousel = ({ recentSearches }) => {
  let searches = recentSearches.map((s) => {
    if (recentSearches.indexOf(s) === 0) {
      return (
        <div className="carousel-item active" key={s.id}>
          <Link to={`/neighborhood/${s.city}`}>
            <img
              src={s.image}
              className="d-block w-100"
              alt={s.city}
              style={{ width: "300px", height: "300px" }}
            />
            <div className="carousel-caption d-block" style={{ top: "8px" }}>
              <h5
                className="bg-primary bg-gradient rounded d-inline p-2"
                style={{ textShadow: "2px 2px black" }}
              >
                {s.city}
              </h5>
            </div>
          </Link>
        </div>
      );
    } else {
      return (
        <div className="carousel-item" key={s.id}>
          <Link to={`/neighborhood/${s.city}`}>
            <img
              src={s.image}
              className="d-block w-100"
              alt={s.city}
              style={{ width: "300px", height: "300px" }}
            />
            <div className="carousel-caption d-block" style={{ top: "8px" }}>
              <h5
                className="bg-primary bg-gradient rounded d-inline p-2"
                style={{ textShadow: "2px 2px black" }}
              >
                {s.city}
              </h5>
            </div>
          </Link>
        </div>
      );
    }
  });

  return (
    <div
      id="carouselExampleControls"
      className="carousel slide"
      data-bs-ride="carousel"
    >
      <div className="carousel-inner">{searches}</div>

      <button
        className="carousel-control-prev"
        type="button"
        data-bs-target="#carouselExampleControls"
        data-bs-slide="prev"
      >
        <span className="carousel-control-prev-icon" aria-hidden="true"></span>
        <span className="visually-hidden">Previous</span>
      </button>
      <button
        className="carousel-control-next"
        type="button"
        data-bs-target="#carouselExampleControls"
        data-bs-slide="next"
      >
        <span className="carousel-control-next-icon" aria-hidden="true"></span>
        <span className="visually-hidden">Next</span>
      </button>
    </div>
  );
};

export default SearchesCarousel;
