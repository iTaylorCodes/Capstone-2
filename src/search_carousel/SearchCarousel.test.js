import SearchesCarousel from "../search_carousel/SearchesCarousel";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";

it("renders without crashing", () => {
  const recentSearches = [
    {
      city: "Los Angeles, CA",
      image:
        "https://a.cdn-hotels.com/gdcs/production65/d1094/007d42a5-84e4-418e-a07d-a020fe670a43.jpg?impolicy=fcrop&w=800&h=533&q=medium",
      id: "test-id",
    },
    {
      city: "San Diego, CA",
      image:
        "https://www.visittheusa.com/sites/default/files/styles/16_9_1280x720/public/images/hero_media_image/2016-11/Hero_San%20Diego%20Skyline_John%20Bahu.jpg?h=3767f04f&itok=R32qq4Na",
      id: "test-id2",
    },
  ];

  render(
    <MemoryRouter>
      <SearchesCarousel recentSearches={recentSearches} />
    </MemoryRouter>
  );
});

it("matches snapshot", () => {
  const recentSearches = [
    {
      city: "Los Angeles, CA",
      image:
        "https://a.cdn-hotels.com/gdcs/production65/d1094/007d42a5-84e4-418e-a07d-a020fe670a43.jpg?impolicy=fcrop&w=800&h=533&q=medium",
      id: "test-id",
    },
    {
      city: "San Diego, CA",
      image:
        "https://www.visittheusa.com/sites/default/files/styles/16_9_1280x720/public/images/hero_media_image/2016-11/Hero_San%20Diego%20Skyline_John%20Bahu.jpg?h=3767f04f&itok=R32qq4Na",
      id: "test-id2",
    },
  ];

  const { asFragment } = render(
    <MemoryRouter>
      <SearchesCarousel recentSearches={recentSearches} />
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});
