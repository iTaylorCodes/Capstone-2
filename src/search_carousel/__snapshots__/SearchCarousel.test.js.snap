// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="carousel slide"
    data-bs-ride="carousel"
    id="carouselExampleControls"
  >
    <div
      class="carousel-inner"
    >
      <div
        class="carousel-item active"
      >
        <a
          href="/neighborhood/Los Angeles, CA"
        >
          <img
            alt="Los Angeles, CA"
            class="d-block w-100"
            src="https://a.cdn-hotels.com/gdcs/production65/d1094/007d42a5-84e4-418e-a07d-a020fe670a43.jpg?impolicy=fcrop&w=800&h=533&q=medium"
            style="width: 300px; height: 300px;"
          />
          <div
            class="carousel-caption d-block"
            style="top: 8px;"
          >
            <h5
              class="bg-primary bg-gradient rounded d-inline p-2"
              style="text-shadow: 2px 2px black;"
            >
              Los Angeles, CA
            </h5>
          </div>
        </a>
      </div>
      <div
        class="carousel-item"
      >
        <a
          href="/neighborhood/San Diego, CA"
        >
          <img
            alt="San Diego, CA"
            class="d-block w-100"
            src="https://www.visittheusa.com/sites/default/files/styles/16_9_1280x720/public/images/hero_media_image/2016-11/Hero_San%20Diego%20Skyline_John%20Bahu.jpg?h=3767f04f&itok=R32qq4Na"
            style="width: 300px; height: 300px;"
          />
          <div
            class="carousel-caption d-block"
            style="top: 8px;"
          >
            <h5
              class="bg-primary bg-gradient rounded d-inline p-2"
              style="text-shadow: 2px 2px black;"
            >
              San Diego, CA
            </h5>
          </div>
        </a>
      </div>
    </div>
    <button
      class="carousel-control-prev"
      data-bs-slide="prev"
      data-bs-target="#carouselExampleControls"
      type="button"
    >
      <span
        aria-hidden="true"
        class="carousel-control-prev-icon"
      />
      <span
        class="visually-hidden"
      >
        Previous
      </span>
    </button>
    <button
      class="carousel-control-next"
      data-bs-slide="next"
      data-bs-target="#carouselExampleControls"
      type="button"
    >
      <span
        aria-hidden="true"
        class="carousel-control-next-icon"
      />
      <span
        class="visually-hidden"
      >
        Next
      </span>
    </button>
  </div>
</DocumentFragment>
`;
